components/AddDonation.js

```diff
@@ -0,0 +1,136 @@
+import styled from "styled-components";
+import { useState } from "react";
+import Counter from "./Counter";
+
+export default function AddDonation({ createDonation }) {
+  const [toggleAddButton, setToggleAddButton] = useState(true);
+  const [count, setCount] = useState(0);
+  function handleClick() {
+    setToggleAddButton(!toggleAddButton);
+  }
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    createDonation(data);
+    setCount(0);
+    event.target.reset();
+    setToggleAddButton(!toggleAddButton);
+  }
+
+  return (
+    <StyledDiv>
+      {toggleAddButton ? (
+        <StyledButton onClick={handleClick}>
+          Neue Spende hinzufügen
+        </StyledButton>
+      ) : (
+        <StyledForm onSubmit={handleSubmit}>
+          <label htmlFor="category">Kategorie:</label>
+          <select name="category" id="category" required>
+            <option value="">-Wähle bitte eine Kategorie-</option>
+            <option value="fruits&vegetables">Obst & Gemüse</option>
+            <option value="homemade">Selbstgemacht</option>
+            <option value="pastry">Backwaren</option>
+            <option value="dairy">Käse, Eier & Molkerei</option>
+            <option value="frozen">Tiefkühlkost</option>
+            <option value="sweet&salty">Süßes & Salziges</option>
+            <option value="beverage">Getränke</option>
+            <option value="meat&fish">Fleisch & Fisch</option>
+          </select>
+          <label htmlFor="description">Beschreibung:</label>
+          <textarea
+            type="text"
+            id="description"
+            name="description"
+            cols="30"
+            rows="5"
+            maxLength="250"
+            pattern="[a-zA-ZäüöÄÜÖß0-9 _-+&*.,:;!?()%€]"
+            onChange={(event) => setCount(event.target.value.length)}
+            required
+          />
+          <Counter max={250} counter={count} />
```

Comment:
At first it was hard to understand what kind of counter this is. If you want to make it perfect you could rename this to TextLengthCounter. But you don't have to.

Reply:
changed it as suggested 🙃